sdks/java/fn-execution/src/main/java/org/apache/beam/sdk/fn/data/BeamFnDataInboundObserver2.java

```diff
@@ -62,6 +62,9 @@ public static BeamFnDataInboundObserver2 forConsumers(
   private final Map<String, Map<String, EndpointStatus<TimerEndpoint<?>>>>
       transformIdToTimerFamilyIdToTimerEndpoint;
   private final CancellableQueue<BeamFnApi.Elements> queue;
+  // Share an exception used for close as constructing exception stack traces is expensive.
+  private static final Exception closeException =
```

Comment:
This will confuse people if we keep the original stack trace since it will be bound to the construction of this instance.

We'll want to create an exception type or find an a suitable alternative that passes false for `writeableStackTrace`:
https://docs.oracle.com/javase/7/docs/api/java/lang/RuntimeException.html#RuntimeException(java.lang.String,%20java.lang.Throwable,%20boolean,%20boolean)

Reply:
Created a subclass